coalib/bearlib/abstractions/Linter.py

```diff
@@ -0,0 +1,680 @@
+                :param file:
+                    The contents of the file.
+                :param config_file:
+                    The path of the config-file if used. ``None`` if unused.
+                :return:
+                    A sequence of arguments to feed the linter-tool with.
+                """
+                raise NotImplementedError
+
+            @staticmethod
+            def get_executable():
+                """
+                Returns the executable of this class.
+
+                :return:
+                    The executable name.
+                """
+                return options["executable"]
+
+            @classmethod
+            def check_prerequisites(cls):
+                """
+                Checks whether the linter-tool the bear uses is operational.
+
+                :return:
+                    True if available, otherwise a string containing more info.
+                """
+                if shutil.which(cls.get_executable()) is None:
+                    return repr(cls.get_executable()) + " is not installed."
+                else:
+                    if options["prerequisite_check_command"]:
+                        try:
+                            check_call(options["prerequisite_check_command"],
+                                       stdout=DEVNULL,
+                                       stderr=DEVNULL)
+                            return True
+                        except (OSError, CalledProcessError):
+                            return options["prerequisite_check_fail_message"]
+                    return True
+
+            @classmethod
+            def _get_create_arguments_metadata(cls):
+                return FunctionMetadata.from_function(
+                    cls.create_arguments,
+                    omit={"filename", "file", "config_file"})
+
+            @classmethod
+            def _get_generate_config_metadata(cls):
+                return FunctionMetadata.from_function(
+                    cls.generate_config,
+                    omit={"filename", "file"})
+
+            @classmethod
+            def _get_process_output_metadata(cls):
+                return FunctionMetadata.from_function(
+                    cls.process_output,
+                    omit={"self", "output", "filename", "file"})
+
+            @classmethod
+            def get_non_optional_settings(cls):
+                return cls.get_metadata().non_optional_params
+
+            @classmethod
+            def get_metadata(cls):
+                merged_metadata = FunctionMetadata.merge(
+                    cls._get_process_output_metadata(),
+                    cls._get_generate_config_metadata(),
+                    cls._get_create_arguments_metadata())
+                merged_metadata.desc = inspect.getdoc(cls)
+                return merged_metadata
+
+            @classmethod
+            def _execute_command(cls, args, stdin=None):
```

Comment:
Do we really need this ? It's a function that calls and returns another function with minor modifications.
And the function is used only once (and I cannot think of any scenario where the bear writer would need to overwrite this)

It just consumes an unnecessary documentation comment which is fairly easy to understand already